src/Quill.php

```diff
@@ -361,9 +375,19 @@ public function registerClientScript()
         $asset->theme = $this->theme;
         
         $configs = Json::encode($this->_quillConfiguration);
-        $editor = 'q_' . \preg_replace('~[^0-9_\p{L}]~u', '_', $this->id);
+        $editor = 'q_' . Inflector::slug($this->id, '_');
+
+        $js = '';
+        if (!empty($this->icons)) {
+            $js .= "var {$editor}_icons = Quill.import('ui/icons');";
+            foreach ($this->icons as $key => $icon) {
+                $icon = Json::encode($icon);
```

Comment:
Why it's JSON encoded?

Reply:
To be sure it will not break js code if some put's strings with quotes oder backslashes or even code injections... `Json::encode` produces from **'test'** => **"test"** and from **'test\\'** => **"Test\\\\"**